### PR TITLE
fix: prevent blocking for setup scripts and cleanup

### PR DESCRIPTION
**Pull request by scissorsneedfoodtoo**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)

Closes #XXXXX

This PR moves scripts to set up i18n, trending links, and Algolia to below the closing `body` tag to prevent render blocking.

It also removes the commented out RunKit script, which were instead added to the individual articles themselves.

### CODE REVIEW

**Comment by ojeytonwilliams** [comment]
Won't these still be blocking, though?  Unless the script has `async` or `defer` the browser will read it before finishing building the DOM.

File: default.hbs [hunk 1]
```diff
@@ -33,22 +33,8 @@
...
-
-    {{!-- Setup global links and index objects for locale --}}
-    <script type="text/javascript" src="{{asset "config/setup-locale.js"}}"></script>
-
-    {{!-- Runkit embeds - NOT WORKING? TODO --}}
-    {{!--
-    <script src="https://embed.runkit.com"></script>
-    --}}
+    <script defer type="text/javascript" src="assets/config/{{@site.lang}}/main.js"></script>
+    <script defer type="text/javascript" src="assets/config/{{@site.lang}}/footer.js"></script>
```
Comment (ojeytonwilliams) [inline default.hbs]: Unfortunately, we'll run into caching problems if we do this.  Ghost gives these `max-age=31536000`, but without the `asset` helper it doesn't generate the cache-busting hash.

Comment (ojeytonwilliams) [reply default.hbs]: For what it's worth, I'm no longer seeing render-blocking JS being flagged up by Lighthouse.  The main remaining issue is the server response, which 11ty should :crossed_fingers: be able to improve.

**Comment by scissorsneedfoodtoo** [comment]
The latest changes seem to solve the render blocking issue, and only load the files for the locale the site is at the moment. The big issue is we can't use the `asset` helper to cache things.

Edit:

> Unfortunately, we'll run into caching problems if we do this. Ghost gives these `max-age=31536000`, but without the `asset` helper it doesn't generate the cache-busting hash.

You beat me to it!

Right, I forgot that we set `max-age` like that.

Ghost [won't serve any JSON files except `manifest.json`](https://forum.ghost.org/t/working-with-static-data/8409/6), and I'm not sure if it's possible to fetch a local JS file.

Maybe the best thing we can do for now is load all the files and defer them. And once the Babel updates from https://github.com/freeCodeCamp/news-theme/pull/109 are merged, we could serve the minified versions of the files instead.

Edit 2:

Another option might be to move these locale specific setup files to our CDN instead and serve them as JSON. Then we could fetch only the files we need when we need them.

**Comment by scissorsneedfoodtoo** [comment]
Sorry for missing this @ojeytonwilliams. Did a rebase with main and force pushed the latest updates.

Is there anything else we should do here? I believe the main changes are still just adding the `defer` attribute to the scripts, and wrapping the code for each in a `DOMContentLoaded` listener.

**Review by ojeytonwilliams — APPROVED** [review]
Sorry for the delay, Kris, this LGTM :+1:

**Review by raisedadead — COMMENTED** [review]
@scissorsneedfoodtoo please deploy

**Comment by scissorsneedfoodtoo** [comment]
Thanks again for the review @ojeytonwilliams. I'll deploy this to the different News instances now.

**Pull request merged**